core/src/main/java/com/datastax/oss/driver/internal/core/addresstranslation/PrivateLinkAddressTranslator.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.addresstranslation;
+
+import com.datastax.oss.driver.api.core.addresstranslation.AddressTranslator;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.api.core.config.DriverOption;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.net.InetSocketAddress;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PrivateLinkAddressTranslator implements AddressTranslator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(PrivateLinkAddressTranslator.class);
+
+  public static final DriverOption ADVERTISED_HOSTNAME_OPTION =
+      () -> "advanced.address-translator.advertised-hostname";
+  private final String advertisedHostname;
+
+  public PrivateLinkAddressTranslator(@NonNull DriverContext context) {
+    DriverExecutionProfile config = context.getConfig().getDefaultProfile();
+    advertisedHostname = config.getString(ADVERTISED_HOSTNAME_OPTION);
+  }
+
+  @NonNull
+  @Override
+  public InetSocketAddress translate(@NonNull InetSocketAddress address) {
+    final int port = address.getPort();
+    LOG.debug(String.format("translating %s:%d to %s:%d", address, port, advertisedHostname, port));
```

Comment:
You should avoid `String.format`:
```suggestion
    LOG.debug("Translating {}:{} to {}:{}", address, port, advertisedHostname, port));
```